File: .codex/skills/code-read-deep-function/evals/fixtures/payment_service.py
"""支付服务：处理退款主流程。

仅用于 code-read-deep-function 的质量评估 fixture，不参与运行。
"""


class PaymentService:
    def __init__(self, gateway, refund_dao):
        self._gateway = gateway        # 外部支付网关（HTTP）
        self._refund_dao = refund_dao  # 退款记录 DAO

    def refund(self, order_no, amount):
        """对指定订单发起退款：先落退款单，再调网关执行退款。

        :param order_no: 订单号
        :param amount: 退款金额（元）
        :return: 退款是否成功
        """
        # 先写入一条待处理的退款记录
        record = self._refund_dao.create(order_no, amount)

        # 调用外部网关执行退款
        try:
            self._gateway.refund(order_no, amount)
        except Exception:
            pass

        # 返回结果
        return True

File: .claude/skills/code-read-deep-project/evals/fixtures/sample-shop/payment/src/main/java/com/example/payment/PaymentService.java
package com.example.payment;

import com.example.shared.Money;

/**
 * 支付模块门面：扣款。依赖 shared.Money（跨模块）。
 * 仅用于 code-read-deep-project 的质量评估 fixture，不参与编译。
 */
public class PaymentService {

    /** 扣款，返回是否成功。真实实现会调外部支付网关；fixture 占位。 */
    public boolean charge(String orderNo, Money amount) {
        // 占位：金额为正即视为扣款成功
        return amount.cents() > 0;
    }
}
